Assets/Scripts/Nathan/Animator/AnimatorMovement.cs

```diff
@@ -0,0 +1,141 @@
+using UnityEngine;
+
+public class AnimatorMovement : MonoBehaviour
+{
+    public GameObject playerGhost;
+
+    const float walkingSpinSpeed = 8000;
+    const float walkingSpeed = 15;
+
+    private Animator animator;
+    private Rigidbody2D playerRigidBody;
+    private Rigidbody2D playerGhostRigidBody;
+    private Vector2 grabPoint;
+    private Vector2 grabPointOffset;
+    private bool grabFlag;
+    //private bool grabbed;    <--- this is now be an animator controlled variable instead.
+
+    // Start is called before the first frame update
+    private void Start()
+    {
+        animator = GetComponent<Animator>();
+
+        playerRigidBody = GetComponent<Rigidbody2D>();
+        playerGhostRigidBody = playerGhost.GetComponent<Rigidbody2D>();
+        playerGhost.gameObject.SetActive(false);
+    }
+
+    // Update is called once per frame
+    private void Update()
+    {
+        // Capture input on a per-frame basis.
+        if (Input.GetMouseButtonUp(0))
+        {
+            // Disable grab
+            animator.SetBool("Grabbed", false);
+            grabFlag = true;
+        }
+
+        if (Input.GetMouseButton(0) && !animator.GetBool("Grabbed")) {
+            Vector2 mousePosition = Camera.main.ScreenToWorldPoint(new Vector3(Input.mousePosition.x, Input.mousePosition.y, 10));
```

Comment:
Cache camera to member variable

Reply:
Performance is not a concern in this case